experimental/ConsoleApp6/ConsoleApp6/Api.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+
+namespace System.Diagnostics
+{
+    public class MetricSource
+    {
+        public MetricSource(string componentName, Version version = default, IDictionary<string,string> defaultLabels = null) { }
+        
+        public Counter CreateCounter(string counterName, IDictionary<string,string> defaultLabels = null)
+        {
+            Metric m = new Metric(this, counterName, MetricAggregations.Sum, defaultLabels);
+            return new Counter(m);
+        }
+
+        public Gauge CreateGauge(string counterName, IDictionary<string, string> defaultLabels = null)
+        {
+            Metric m = new Metric(this, counterName, MetricAggregations.Sum, defaultLabels);
+            return new Gauge(m);
+        }
+    }
+
+
+    //TODO: do we have different metric types or a generic parameter to allow collection of different integer/float sizes?
+    public class Metric
+    {
+        internal Metric(MetricSource source,
+                        string name,
+                        MetricAggregation defaultAggregation,
+                        IDictionary<string, string> defaultLabels)
+        {
+            Source = source;
+            Name = name;
+            DefaultAggregation = defaultAggregation;
+            DefaultLabels = new LabelSet(defaultLabels);
+        }
+
+        public MetricSource Source { get; }
+        public string Name { get; }
+        public MetricAggregation DefaultAggregation { get; }
+        public LabelSet DefaultLabels { get; }
+
+        public void Record(int measurement) { }
+        public void Record(int measurement, string labelName1, string labelValue1) { }
```

Comment:
Looks like you're separating Key and Value.  So, why not Record(..., string[] keys, string[] values)?

Reply:
Sorry I completely missed some of these comments. My bad!
A few reasons:
a) It is one less allocation per metric recording (cost ~10ns)
b) passing strings at Record implies the names can be different every time which means the SDK has to search and/or sort them

Reply:
RE: b.  Although I agree, I don't think that's your top priority concern.